### PR TITLE
'EditAccount' screen

### PR DESCRIPTION
**Pull request by KassFlute**
I implemented the ability for the users to edit their own profile in a dedicated screen "EditAccount.kt".

 - The edited informations are only saved in the db when the "save" button is pressed while the "back" button discard them.
 - I added a security to not be able to edit other's profile (even if the edit button should only be displayed on user's own profile)

### CODE REVIEW

**Review by LozanoCyrielle — CHANGES_REQUESTED** [review]
Good job! Just small details :

-  you have unused variable (emptyMockUiState) in your code 
- you mock the entire navigation when you only need the goBack.

**Comment by KassFlute** [comment]
Good points, I've removed these useless elements and an additional useless import.

File: app/src/main/java/com/android/partagix/model/UserViewModel.kt [hunk 1]
```diff
@@ -49,8 +49,7 @@ class UserViewModel(
 
     if (userID == null) {
       database.getUser("XogPd4oF1nYc6Rag6zhh") { updateUIState(it) }
-    } else if (userID != "" &&
-        false) { // TODO: remove false when logged in users are in the database
+    } else if (userID != "") {
       database.getUser(userID) { updateUIState(it) }
     } else {
       database.getUser("XogPd4oF1nYc6Rag6zhh") { updateUIState(it) }
```
Comment (Valentin889) [inline app/src/main/java/com/android/partagix/model/UserViewModel.kt:55]: Seems like we'll never go to this else so I would remove it

**Review by Valentin889 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/android/partagix/model/UserViewModel.kt (hunk 1 above)
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/model/UserViewModel.kt:53]: You're right ! I've changed the if-else condition to make it clearer in  `setUserToCurrent()`

**Comment by LozanoCyrielle** [comment]
I just realized that you cannot use the friend button as edit, because we will need it later

**Comment by KassFlute** [comment]
Nice, thank you for adding a button. I thought the friend button (that was called 'edit' xD) was depreciated and could be replaced by the edit button.

**Review by LozanoCyrielle — APPROVED** [review]
Now everything looks good to me. Nice work :))

**Review by Valentin889 — APPROVED** [review]
Seems good to me

**Pull request merged**